autogen/coding/ipython_code_executor.py

```diff
@@ -50,15 +54,22 @@ class UserCapability:
 the output will be a path to the image instead of the image itself.
 """
 
-        def add_to_agent(self, agent):
+        def add_to_agent(self, agent: Agent) -> None:
```

Comment:
agent.system_message can be either a string or a list of strings.

Reply:
Perhaps create a protocol called `LLMAgent` with the `update_system_message` method? This can be used inplace of `Agent` whenever we need to use `add_to_agent`.

Reply:
yes

Reply:
changed